docs/core/diagnostics/metrics-collection.md

```diff
@@ -0,0 +1,404 @@
+instrument is created within the app. Our delegate can examine the instrument, such as checking the name, the Meter, or any other
+public property to decide whether to subscribe. If we do want to receive measurements from this instrument then we invoke
+<xref:System.Diagnostics.Metrics.MeterListener.EnableMeasurementEvents> to indicate that. If your code has another way
+to obtain a reference to an instrument it is legal to invoke EnableMeasurementEvents() at any time with that reference, but this is
+probably uncommon.
+
+```C#
+meterListener.SetMeasurementEventCallback<int>(OnMeasurementRecorded);
+...
+static void OnMeasurementRecorded<T>(Instrument instrument, T measurement, ReadOnlySpan<KeyValuePair<string,object>> tags, object? state)
+{
+    Console.WriteLine($"{instrument.Name} recorded measurement {measurement}");
+}
+```
+Next we configured the delegate that is invoked when measurements are received from an instrument by calling
+<xref:System.Diagnostics.Metrics.MeterListener.SetMeasurementEventCallback>. The generic parameter controls which data type
+of measurement will be received by the callback. For example a `Counter<int>` generates `int` measurements whereas a
+`Counter<double>` will generate `double` measurements. Instruments are allowed to be created with `byte`, `short`, `int`, `long`,
+`float`, `double`, and `decimal` types. We recommend registering a callback for every data type unless you have scenario
+specific knowledge that not all data types will be needed, such as in this example. Making repeated calls to
+SetMeasurementEventCallback() with different generic arguments may appear a little unusual. The API is designed this way
+to allow MeterListeners to receive measurements with extremely low performance overhead, typically just a few ns on the
+recent .NET versions.
+
+When MeterListener.EnableMeasurementEvents() was called initially there was an oportunity to provide a `state` object as
+one of the parameters and that object can be anything you want. If you provide a state object in that call then it will be
+stored with that instrument and returned to you as the `state` parameter in the callback. This is intended both as a
+convenience and as a performance optimization. Often listeners need to create an object for each instrument that will
+store measurements in memory and have code to do calculations on those measurements. Although you could create a Dictionary
+that maps from the instrument to the storage object and look it up on every measurement that would be much slower than
+accessing it from `state`.
+
+
+```C#
+meterListener.Start();
+```
+Once the MeterListener is configured we need to start it which triggers callbacks to begin. The InstrumentPublished 
+delegate will be invoked for every pre-existing Instrument in the process. In the future any newly created Instrument
+will also trigger InstrumentPublished to be created.
+
+```C#
+meterListener.Dispose();
```

Comment:
you have used `using ` when explaining the code here. we need to ensure we are consistent if we are using `using` or we manually calling `Dispose`

Reply:
The code and explanation should now be consistent on the 'using' pattern.